contrib/pkg/clusterpool/clusterpool.go

```diff
@@ -0,0 +1,464 @@
+package clusterpool
+
+import (
+	"fmt"
+	"os"
+	"os/user"
+	"path/filepath"
+
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/kubernetes/scheme"
+
+	"github.com/openshift/hive/contrib/pkg/utils"
+	awsutils "github.com/openshift/hive/contrib/pkg/utils/aws"
+	azureutils "github.com/openshift/hive/contrib/pkg/utils/azure"
+	gcputils "github.com/openshift/hive/contrib/pkg/utils/gcp"
+	"github.com/openshift/hive/pkg/apis"
+	hivev1 "github.com/openshift/hive/pkg/apis/hive/v1"
+	hivev1aws "github.com/openshift/hive/pkg/apis/hive/v1/aws"
+	hivev1azure "github.com/openshift/hive/pkg/apis/hive/v1/azure"
+	hivev1gcp "github.com/openshift/hive/pkg/apis/hive/v1/gcp"
+	"github.com/openshift/hive/pkg/constants"
+)
+
+const (
+	cloudAWS   = "aws"
+	cloudAzure = "azure"
+	cloudGCP   = "gcp"
+
+	longDesc = `
+OVERVIEW
+The hiveutil clusterpool create-pool command generates and applies the
+artifacts needed to create a new Hive cluster pool.
+Currently cluster pools are supported in AWS, Azure and GCP.
+
+IMAGES
+An existing ClusterImageSet can be specified with the --image-set
+flag. Otherwise, one will be generated using the images specified for the
+cluster pool. If you don't wish to use a ClusterImageSet, specify
+--use-image-set=false. This will result in images only specified on the
+cluster itself.
+
+ENVIRONMENT VARIABLES
+The command will use the following environment variables for its output:
+
+PULL_SECRET - If present, it is used as the cluster pool's pull
+secret and will override the --pull-secret flag. If not present, and
+the --pull-secret flag is not specified, then the --pull-secret-file is
+used. That file's default value is %[1]s.
+
+RELEASE_IMAGE - Release image to use to install the clusters in the clusterpool.
+If not specified, the --release-image flag is used. If that's not specified, a 
+default image is obtained from a the following URL:
+https://openshift-release.svc.ci.openshift.org/api/v1/releasestream/4-stable/latest
+`
+)
+
+var (
+	validClouds = map[string]bool{
+		cloudAWS:   true,
+		cloudAzure: true,
+		cloudGCP:   true,
+	}
+)
+
+type ClusterPoolOptions struct {
+	Name               string
+	Namespace          string
+	BaseDomain         string
+	PullSecret         string
+	PullSecretFile     string
+	Cloud              string
+	CredsFile          string
+	CloudSecretName    string
+	createCloudSecret  bool
+	ClusterImageSet    string
+	ReleaseImage       string
+	ReleaseImageSource string
+	UseClusterImageSet bool
```

Comment:
This is not used in generating the ClusterPool and is not supported by ClusterPool.